app/src/main/java/com/duckduckgo/app/referral/AppInstallationReferrerStateListener.kt

```diff
@@ -155,11 +161,17 @@ class PlayStoreAppReferrerStateListener @Inject constructor(
     private fun referralResultReceived(result: ParsedReferrerResult) {
         referralResult = result
 
-        if (result is ReferrerFound) {
-            variantManager.updateAppReferrerVariant(result.campaignSuffix)
-            appReferrerDataStore.campaignSuffix = result.campaignSuffix
-
+        when (result) {
+            is CampaignReferrerFound -> {
+                variantManager.updateAppReferrerVariant(result.campaignSuffix)
+                appReferrerDataStore.campaignSuffix = result.campaignSuffix
+            }
+            is EuAuctionReferrerFound -> {
+                variantManager.updateAppReferrerVariant(VariantManager.RESERVED_EU_AUCTION_VARIANT)
```

Comment:
Is this problematic? We would have already made the exti call with the OLD variant so our retention stats for both that and this variant will be incorrect.

Reply:
It shouldn't happen because `AtbInitializer` guards against making the `exti` call before referrer detection is finished.